From 673dc5f525c9d478fc22f8ea0a50d9849a81f6c8 Mon Sep 17 00:00:00 2001
From: Kateryna Kostiuk <kateryna.kostiuk@savoirfairelinux.com>
Date: Fri, 2 Sep 2022 13:32:10 -0400
Subject: [PATCH] conversation: fix crash when block contact

This patch fixes crash that happened when block contact
for not active account.

GitLab: #758
Change-Id: I5347394a67cdffe0d95c9ee03aedf9d2618cec55
---
 src/jamidht/jamiaccount.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/daemon/src/jamidht/jamiaccount.cpp b/daemon/src/jamidht/jamiaccount.cpp
index fe15eedb2..9d0a5ee68 100644
--- a/daemon/src/jamidht/jamiaccount.cpp
+++ b/daemon/src/jamidht/jamiaccount.cpp
@@ -1129,7 +1129,9 @@ JamiAccount::loadAccount(const std::string& archive_password,
                     if (auto convModule = shared->convModule())
                         convModule->removeContact(uri, banned);
                     // Remove current connections with contact
-                    shared->connectionManager_->closeConnectionsWith(uri);
+                    if (shared->connectionManager_) {
+                        shared->connectionManager_->closeConnectionsWith(uri);
+                    }
                     // Update client.
                     emitSignal<DRing::ConfigurationSignal::ContactRemoved>(shared->getAccountID(),
                                                                            uri,
-- 
2.37.3

